app/scenes/channel/channel.js

```diff
@@ -47,21 +50,13 @@ export default class Channel extends React.PureComponent {
         renderRightComponent: (props, emitter) => {
             return <ChannelMenuButton emitter={emitter}/>;
         }
-    }
-
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            leftSidebarOpen: false,
-            rightSidebarOpen: false
-        };
-    }
+    };
 
     componentWillMount() {
         this.props.subscribeToHeaderEvent('open_channel_drawer', this.openChannelDrawer);
         this.props.subscribeToHeaderEvent('open_right_menu', this.openRightMenuDrawer);
         this.props.subscribeToHeaderEvent('show_channel_info', this.props.actions.goToChannelInfo);
+        EventEmitter.on('leave_team', this.handleLeaveTeam);
```

Comment:
What are your thoughts on tying this into the navigation router? Seems like I'm just repeating your better implementation of the event emitter. It could then be passed down to each scene from the router. Thoughts?